algorithms/kernel/k_nearest_neighbors/kdtree_knn_classification_predict_dense_default_batch_impl.i

```diff
@@ -376,23 +307,16 @@ void KNNClassificationPredictKernel<algorithmFpType, defaultDense, cpu>::findNea
                 DAAL_PREFETCH_READ_T0(nx);
                 DAAL_PREFETCH_READ_T0(nx + 16);
 
-                for (i = 0; i < end - start; ++i)
-                {
-                    distance[i] += (query[j - 1] - dx[i]) * (query[j - 1] - dx[i]);
-                }
+                distance_comp<cpu, algorithmFpType>(dx, query[j - 1], distance, length);
 
                 const_cast<NumericTable &>(data).releaseBlockOfColumnValues(xBD[curBDIdx]);
 
-                const auto tempBDIdx = curBDIdx;
-                curBDIdx             = nextBDIdx;
-                nextBDIdx            = tempBDIdx;
+                daal::services::internal::swap<cpu, size_t>(curBDIdx, nextBDIdx);
```

Comment:
Do you specified size_t as a template argument explicitly to perform some conversion?

Reply:
It's just explicit specialization. It affects nothing.